src/app/pages/DashboardPage/RoseChartCard.tsx

```diff
@@ -0,0 +1,89 @@
+import { FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import { ChartDuration, chartDurationToDaysMap } from '../../utils/chart-utils'
+import Box from '@mui/material/Box'
+import { useTheme } from '@mui/material/styles'
+import useMediaQuery from '@mui/material/useMediaQuery'
+import { LineChart } from '../../components/charts/LineChart'
+import { SnapshotCard } from './SnapshotCard'
+import { useGetRosePrice, useGetRoseMarketChart } from '../../../coin-gecko/api'
+
+type RoseChartCardProps = {
+  chartDuration: ChartDuration
+}
+
+type ChartDataType = {
+  timestamp: number
+  value: number
+}
+
+const formatParams = {
+  value: {
+    currency: 'USD',
+    maximumFractionDigits: 5,
+  },
+}
+
+export const RoseChartCard: FC<RoseChartCardProps> = ({ chartDuration }) => {
+  const { t } = useTranslation()
+  const theme = useTheme()
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'))
+  const rosePriceQuery = useGetRosePrice()
+  const roseMarketChartQuery = useGetRoseMarketChart({
+    days: chartDurationToDaysMap[chartDuration],
+    interval:
+      chartDurationToDaysMap[chartDuration] >= chartDurationToDaysMap[ChartDuration.WEEK]
+        ? 'daily'
+        : undefined,
```

Comment:
This seems difficult to understand. Add a comment what undefined interval results in

Reply:
and maybe move this code into api so it is near `index % 12 == 0`